src/D2L.CodeStyle.Analyzers/Language/AwaitedTasksAnalyzer.Codefix.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace D2L.CodeStyle.Analyzers.Language {
+	partial class AwaitedTasksAnalyzer {
+
+		[ExportCodeFixProvider(
+			LanguageNames.CSharp,
+			Name = nameof( ConfigureAwaitedTaskCodeFix )
+		)]
+		public sealed class ConfigureAwaitedTaskCodeFix : CodeFixProvider {
+
+			public override ImmutableArray<string> FixableDiagnosticIds
+				=> ImmutableArray.Create( Diagnostics.AwaitedTaskNotConfigured.Id );
+
+			public override FixAllProvider GetFixAllProvider()
+				=> WellKnownFixAllProviders.BatchFixer;
+
+			public override async Task RegisterCodeFixesAsync(
+				CodeFixContext ctx
+			) {
+				var root = await ctx.Document
+					.GetSyntaxRootAsync( ctx.CancellationToken )
+					.ConfigureAwait( false );
+
+				bool useSafeAsync = await CanReferenceSafeAsync( ctx )
+					.ConfigureAwait( false );
+
+				foreach( var diagnostic in ctx.Diagnostics ) {
+					var span = diagnostic.Location.SourceSpan;
+
+					SyntaxNode syntax = root.FindNode( span, getInnermostNodeForTie: true );
+					if( !( syntax is AwaitExpressionSyntax awaitExpression ) ) {
+						continue;
+					}
+
+					ctx.RegisterCodeFix(
+						CodeAction.Create(
+							title: "Configure awaited task",
+							ct => ConfigureAwaitedTask(
+								orig: ctx.Document,
+								root: root,
+								awaitExpression: awaitExpression,
+								useSafeAsync: useSafeAsync,
+								cancellationToken: ct
+							)
+						),
+						diagnostic
+					);
+				}
+			}
+
+			private static InvocationExpressionSyntax ConfigureTask( ExpressionSyntax task ) =>
+				SyntaxFactory
+					.InvocationExpression( SyntaxFactory.MemberAccessExpression(
+						SyntaxKind.SimpleMemberAccessExpression,
+						task,
+						SyntaxFactory.IdentifierName( "ConfigureAwait" )
+					) )
+					.WithArgumentList( SyntaxFactory.ArgumentList( SyntaxFactory.SingletonSeparatedList(
+						SyntaxFactory
+							.Argument( SyntaxFactory.LiteralExpression( SyntaxKind.FalseLiteralExpression ) )
+							.WithNameColon( SyntaxFactory.NameColon( "continueOnCapturedContext" ) )
+					) ) );
+
+			private static InvocationExpressionSyntax SafeAsyncify( ExpressionSyntax task ) =>
+				SyntaxFactory
+					.InvocationExpression( SyntaxFactory.MemberAccessExpression(
+						SyntaxKind.SimpleMemberAccessExpression,
+						task,
+						SyntaxFactory.IdentifierName( "SafeAsync" )

```

Comment:
There isn't a `SafeAsync` for `ValueTask` yet. Will add to D2L.Core

Reply:
Err, forgot ValueTask is in a nuget package and messing with such dependencies is awful, soooo

Probably not going to see many ValueTasks in the wild still, so going to pretend this isn't a problem.

Reply:
Could we just not use `SafeAsync`? Like if we just required `ConfigureAwait( continueOnCapturedContext: false )` (i.e. including the argument name) then

1) no references to d2l DLLs/vanilla C#
2) all the "safety"
3) its more clear what its doing

Reply:
Are you saying that's something you want to do _now_? If we're going to do that, I think we should switch existing usages as well and then remove the extension methods if we're going to do that.

In our current state, I'd prefer the codefix to produce code that approximates existing code.

```
osmith@KLX1-OSMITH MINGW64 /c/d2l/full/checkout (master)
$ git grep 'ConfigureAwait' | wc -l
84

osmith@KLX1-OSMITH MINGW64 /c/d2l/full/checkout (master)
$ git grep 'SafeAsync' | wc -l
415

osmith@KLX1-OSMITH MINGW64 /c/d2l/full/checkout (master)
$ git grep 'SafeWait' | wc -l
103
```

Reply:
Nah, just it would be cool in the future.